extensions/notebook/src/jupyter/jupyterServerManager.ts

```diff
@@ -105,6 +105,7 @@ export class LocalJupyterServerManager implements nb.ServerManager, vscode.Dispo
 	private async doStartServer(): Promise<IServerInstance> { // We can't find or create servers until the installation is complete
 		let installation = this.options.jupyterInstallation;
 		await installation.promptForPythonInstall();
+		await installation.promptForPackageUpgrade();
```

Comment:
In terms of perf, how much does this impact things? Should we restrict this check to only once per ADS session?

Reply:
We only do an install if there are packages missing. Otherwise the only perf overhead is from doing a "pip/conda list" call, and then checking if our required packages are present in those lists (converted to a Map here for better perf).